docs/docs/tutorial.md

```diff
@@ -499,21 +499,20 @@ Let's make the form interactive. When the user submits the form, we should clear
 var CommentForm = React.createClass({
   handleSubmit: function(e) {
     e.preventDefault();
-    var author = this.refs.author.value.trim();
-    var text = this.refs.text.value.trim();
+    var author = this.authorInput.value.trim();
+    var text = this.textInput.value.trim();
     if (!text || !author) {
       return;
     }
     // TODO: send request to the server
-    this.refs.author.value = '';
-    this.refs.text.value = '';
-    return;
+    this.authorInput.value = '';
+    this.textInput.value = '';
   },
   render: function() {
     return (
       <form className="commentForm" onSubmit={this.handleSubmit}>
-        <input type="text" placeholder="Your name" ref="author" />
-        <input type="text" placeholder="Say something..." ref="text" />
+        <input type="text" placeholder="Your name" ref={(ref) => {this.authorInput = ref}} />
```

Comment:
Usually we recommend that component state be stored using `this.setState()` because that ensures that a rerender will be triggered and the component's rendered value will always equal the component's internal state.